tests/test_product.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+from buy_together_app.models import Product
+
+QR = "Q5o76MbdiNbXprNEnHfpcGWFp1CMF8XY"
+NAME = "Banana"
+DESCRIPTION = "Good!"
+
+
+@pytest.fixture
+def product0():
+    return Product(qr_code=QR, product_name=NAME, description=DESCRIPTION)
+
+
+@pytest.fixture
+def saved_product0(product0):
+    product0.save()
+    return product0
+
+
+test_data = [
+            Product('5yMcE3UrEEmZBukWZBrlCKnu0K5j24UY', 'Apple', 'Sweety!'),
+            Product('Q5o76MbdiNbXprNEnHfpcGWFp1CMF8ad', 'Banana', "Good!"),
+            Product(qr_code=QR, product_name=NAME, description=DESCRIPTION),
+        ]
+
+
+class TestProductModel:
+    @pytest.mark.django_db()
+    def test_save_product(self, saved_product0):
+        assert saved_product0 in Product.objects.all()
+
+    @pytest.mark.django_db()
+    def test_delete_product(self, saved_product0):
+        assert saved_product0 in Product.objects.all()
+        saved_product0.delete()
+        assert saved_product0 not in Product.objects.all()
+
+    @pytest.mark.django_db()
+    def test_none_feilds(self, saved_product0):
+        assert saved_product0.qr_code is not None
+        assert saved_product0.product_name is not None
+        assert saved_product0.description is not None
+
+    @pytest.mark.django_db()
+    def test_filter_by_qr(self, saved_product0):
+        assert [saved_product0] == list(Product.filter_qr(saved_product0.qr_code))
+
+    @pytest.mark.django_db()
+    def test_filter_by_product_name(self, saved_product0):
+        assert saved_product0 in list(Product.filter_name(saved_product0.product_name))
+
+    @pytest.mark.django_db()
+    def test_filter_by_description(self, saved_product0):
+        assert saved_product0 in Product.filter_description(saved_product0.description)
+
+    @pytest.mark.django_db()
+    def test_get_all_products(self, saved_product0):
+        for i in test_data:
+            assert i in Product.objects.all()
```

Comment:
You're not testing `get_all_products` here. You're testing what `get_all_products` is calling...

Reply:
done